lib/pure/collections/sequtils.nim

```diff
@@ -625,20 +733,22 @@ template foldl*(sequence, operation: untyped): untyped =
 template foldl*(sequence, operation, first): untyped =
```

Comment:
I know you didn't touch this, but I always wondered, why this template doesn't have this signature:
``template foldl*[T,U](sequence: openarray[T], operation: proc(T,U):U , first: U): untyped``. I guess it is because that would constrain the calling convention of `operation` as well.

Reply:
As far as I can tell, this is because `operation` is dirty in order to make sure to inline it, as in

```nim
let s = foldl(@[1, 2, 3], a + b, 0)
```

which is **not** the same as

```nim
let s = foldl(@[1, 2, 3], proc(a, b: int): int = a + b, 0)
```

as the latter involves (potentially, unless the compiler does some inlining of its own) a function call for each element